### ISSUE TITLE
linter: default plugins re-enabled when a config extends a base that sets "plugins"

### ISSUE THREAD

**Issue by FarisTF** [oxc-project/oxc#24836]
**What version of Oxlint are you using?**

1.75.0

**What command did you run?**

`npx oxlint -c leaf.json probe.js`

**What does your config file look like?**

```jsonc
// base.json
{
  "plugins": ["react"]
}
```

```jsonc
// leaf.json
{
  "extends": ["./base.json"]
}
```

```js
// probe.js
export function f(a) {
  return { ...(a || {}) }
}
```

**What happened?**

The run reports `unicorn(no-useless-fallback-in-spread)`, so the default plugins are active even though the extended base sets `"plugins": ["react"]`. The [nested config docs](https://oxc.rs/docs/guide/usage/linter/nested-config.html) list `plugins` as one of the properties `extends` supports, so I expected the leaf to inherit the base's list.

Two variations behave correctly:

1. Adding `"plugins": ["react"]` to leaf.json itself: no unicorn diagnostics.
2. A single config with `"plugins": ["react"]` and no `extends`: no unicorn diagnostics (the behavior #6896 / #7303 established).

So an explicit `plugins` list replaces the defaults in a single file, but does not survive `extends` when the extending config omits its own `plugins` key.

I hit this while centralizing oxlint config for a 5-app pnpm monorepo, where the shared base intentionally enables fewer plugins than the defaults for eslint-parity reasons. Every consuming app currently re-declares `plugins` as a workaround.

Is this intended (defaults acting as a per-file floor) or a gap in the extends merge? If it's intended I'm happy to send a docs PR spelling it out; if it's a bug I'd like to try a fix.

I used Claude Code to help investigate and draft this; I verified the repro and both control cases by hand on 1.75.0.

**Comment by FarisTF** [comment]
If this is confirmed as a bug rather than intended behavior, I'd like to attempt the fix. Pointers to where the extends merge handles `plugins` would be appreciated.

**Comment by FarisTF** [comment]
Dug into this myself. It looks intended: `test_extends_plugins` in `crates/oxc_linter/src/config/config_builder.rs` asserts exactly this behavior. Test 3's comment says a child without `plugins` "should inherit parent's plugins and apply the default plugins from the child config", and Test 9 asserts the same defaults-union in the reverse direction.

Test 8 also shows a supported way to get pure inheritance that I couldn't find documented: `"plugins": []` in the extending config unions with the base's list without re-adding the defaults. Verified on 1.75.0 against my monorepo setup; it solves my case.

So the model seems to be: default plugins are a per-config floor, `extends` unions plugin sets, and an explicit `plugins` array (even `[]`) opts out of the floor. If that reading is right, the gap here is documentation, not behavior: the nested config page currently only says plugins can be extended.

Happy to send a docs PR to the website (nested-config.md) documenting the floor and the `[]` idiom, and to close this issue if you agree it's working as designed.

I used Claude Code to help investigate; the test reading and the `plugins: []` behavior were verified by hand on 1.75.0.

**Comment by FarisTF** [comment]
Docs merged in oxc-project/website#1228, so closing as working-as-designed. `"plugins": []` is the idiom for exact inheritance.
